### PR TITLE
Mention discrepency between original impl and our impl of RAdam

### LINKED ISSUE #88836: Weight decay handled differently in torch.optim.radam compared to original implementation
### 🐛 Describe the bug

Comparing the implementation in torch.optim.radam with https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py , I noticed that the original author's implementation applies weight decay at the very end as a decoupled step, whereas the torch.optim.radam implementation applies weight decay immediately. The difference is significant because only in the latter does weight decay affect the exponential average computation. 

Also worth noting, the Radam paper specifically mentions using the decoupled weight norm formulation by citing the AdamW paper in the first paragraph of the experiments section.

### Versions

Based on the radam implementation in the current master branch.

cc @vincentqb @jbschlosser @albanD

### PR DESCRIPTION
**Pull request by albanD**
Fixes https://github.com/pytorch/pytorch/issues/88836

### CODE REVIEW

File: torch/optim/radam.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ class RAdam(Optimizer):
 
     For further details regarding the algorithm we refer to `On the variance of the adaptive learning rate and beyond`_.
 
+    This implementation uses the same weight_decay implementation as Adam (were the weight_decay is applied
+    to the gradient) and not the one from AdamW (were weight_decay is applied to the update). Note that, for
+    historical reasons, this is different from the `original author implementation`_.
```
Comment (mruberry) [inline torch/optim/radam.py:49]: "author's implementation"?

**Review by mruberry — APPROVED** [review]
Stamped! See small suggestion

File: torch/optim/radam.py (hunk 1 above)
Comment (albanD) [reply torch/optim/radam.py:49]: Ho yes I had that before and then the link didn't work and I wasn't sure if it was because of the `'` or not. I should be able to add it back.

[the remark this answers is not in the document]
Comment (albanD) [reply torch/optim/radam.py:49]: I'm not sure tbh. It does match the other optimizers and is not specifically specified in the paper. So I guess that this was the default implementation we came up with.
But I don't know for sure so I left this vague. You think I should investigate the history to figure this out?

[the remark this answers is not in the document]
Comment (mruberry) [reply torch/optim/radam.py]: I imagine you have more valuable things to do, but "historical reasons" is so vague I wonder if it's worth mentioning at all? This could just say, "This is different from..."
